ocp_resources/network_attachment_definition.py

```diff
@@ -197,3 +197,39 @@ def to_dict(self):
     @property
     def resource_name(self):
         return f"ovs-cni.network.kubevirt.io/{self.bridge_name}"
+
+
+class OVNOverlayNetworkAttachmentDefinition(NetworkAttachmentDefinition):
+    def __init__(
+        self,
+        network_name=None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Create and manage an OVN k8s overlay NetworkAttachmentDefinition (a switched, layer 2, topology network).
+
+        API reference:
+        https://docs.openshift.com/container-platform/4.13/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-network-plugin-json-object_configuring-additional-network
+
+        Args:
+            network_name (str, default: None): The name of the network. This field is mandatory when not using yaml
+                file.
+        """
+        super().__init__(
+            cni_type="ovn-k8s-cni-overlay",
+            *args,
+            **kwargs,
+        )
+        self.network_name = network_name
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.network_name:
```

Comment:
If user send `yaml_file` and not  `network_name` I will fail while it shouldn't

Reply:
Ack.